src/test/java/uk/gov/pay/api/it/fixtures/PaymentResultBuilder.java

```diff
@@ -0,0 +1,228 @@
+        }
+        
+        public RefundSummary(String status, long amountAvailable, long amountSubmitted) {
+            this.status = status;
+            this.amount_available = amountAvailable;
+            this.amount_submitted = amountSubmitted;
+            this.user_external_id = null;
+        }
+    }
+
+    protected static class SettlementSummary {
+        public String capture_submit_time;
+        public String captured_date;
+        
+        public SettlementSummary() {}
+        
+        public SettlementSummary(uk.gov.pay.api.model.SettlementSummary settlementSummary) {
+            this.capture_submit_time = settlementSummary.getCaptureSubmitTime();
+            this.captured_date = settlementSummary.getCapturedDate();
+        }
+        
+        public SettlementSummary(String captureSubmitTime, String capturedDate) {
+            this.capture_submit_time = captureSubmitTime;
+            this.captured_date = capturedDate;
+        }
+    }
+
+    protected static class TestPayment {
+        public TestPaymentState state;
+        public String charge_id, description, reference, email, created_date;
+        public long amount;
+        public String gateway_transaction_id, return_url, payment_provider;
+        public String language;
+        public boolean delayed_capture;
+        public RefundSummary refund_summary;
+        public SettlementSummary settlement_summary;
+        public CardDetails card_details = new CardDetails();
+        public Long corporate_card_surcharge, total_amount;
+        public List<ImmutableMap<?,?>> links;
+    }
+
+    protected static class TestPaymentState {
+        public String status;
+        public boolean finished;
+
+        protected TestPaymentState(String status, boolean finished) {
+            this.status = status;
+            this.finished = finished;
+        }
+    }
+
+    protected static class TestPaymentSuccessState extends TestPaymentState {
+        private boolean success;
+
+        protected TestPaymentSuccessState(String status) {
+            super(status, true);
+            this.success = true;
+        }
+    }
+
+    protected static class TestPaymentFailureState extends TestPaymentState {
+        private boolean success;
+        private String message, code;
+
+        protected TestPaymentFailureState(String status, String message, String code) {
+            super(status, true);
+            this.success = false;
+            this.message = message;
+            this.code = code;
+        }
+    }
+
+    protected static final List<TestPaymentState> states = new LinkedList<>();
+
+    static {
+        states.add(new TestPaymentState("created", false));
+        states.add(new TestPaymentState("started", false));
+        states.add(new TestPaymentState("submitted", false));
+        states.add(new TestPaymentSuccessState("success"));
+    }
+
+    protected long amount = DEFAULT_AMOUNT;
+    protected String reference = null;
```

Comment:
Not sure why some of these are set to `null` and others aren’t but I see the old code did this too